monitor/server.cpp

```diff
@@ -215,33 +222,45 @@ uint64_t handler::parse_request(size_t size) {
 		{"/top", DNET_MONITOR_TOP}
 	};
 
-	const char *packet = m_buffer.data();
-	const char* end = packet + size;
-	const char *method_end = std::find(packet, end, ' ');
-	if (method_end >= end || packet == method_end)
-		return 0;
-
-	const char *url_begin = method_end + 1;
-	const char *url_end = std::find(url_begin, end, ' ');
-	if (url_end >= end)
-		return 0;
+	request req;
 
-	m_url = std::string(url_begin, url_end);
-
-	auto it = handlers.find(m_url);
+	auto it = handlers.find(m_http_request.path());
 	if (it != handlers.end()) {
-		return it->second;
-	} else if (ssize_t(categories_url.size()) < (url_end - url_begin) &&
-	           strncmp(url_begin, categories_url.c_str(), categories_url.size()) == 0) {
-		const char *categories = url_begin + categories_url.size();
-		try {
-			return std::stoull(std::string(categories, url_end));
-		} catch(...) {
-			DNET_LOG_ERROR(m_monitor.node(), "Can't parse categories: {}", categories);
+		req.categories = it->second;
+	} else if (m_http_request.path() == "/") {
+		auto it = m_http_request.query().find("categories");
+		if (it != m_http_request.query().end()) {
+			try {
+				req.categories = std::stoull(it->second);
+			} catch (...) {
+				DNET_LOG_ERROR(m_monitor.node(), "monitor: http-server: Can't parse categories: {}",
+				               it->second);
+				return request();
+			}
+		}
+	} else {
+		DNET_LOG_ERROR(m_monitor.node(), "monitor: http-server: request parser: Unknown path: {}",
+		               m_http_request.path());
+		return request();
+	}
+
+	const auto backends_item = m_http_request.query().find("backends");
+	if (backends_item != m_http_request.query().end()) {
+		std::string id_list = backends_item->second;
+		char *id = std::strtok(const_cast<char*>(id_list.c_str()), ",");
+		while (id != NULL) {
+			try {
+				req.backends_ids.insert(std::strtoul(id, nullptr, 10));
+			} catch (...) {
+				DNET_LOG_ERROR(m_monitor.node(), "monitor: http-server: Can't parse backend ids: {}",
+				               id_list);
```

Comment:
`std::strok` modifies `id_list`, so here `backends_item->second` should be used.